app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/MessageMapEntitiesViewModel.java

```diff
@@ -29,32 +32,29 @@ public void onEntityClick(Entity entity) {
 
     public Entity addMessage(Message message) {
         PointGeometry point = getPointGeometry(message);
+        Symbol symbol = mSymbolizer.symbolize(message);
 
-        Entity userEntity = createMessagePinEntity(point);
+        Entity userEntity = createMessagePinEntity(point, symbol);
         mEntityToMessageHashMap.put(userEntity, message);
         userEntity.setOnClickListener(this);
         return userEntity;
     }
 
     private PointGeometry getPointGeometry(Message message) {
         switch (message.getType()) {
-            case Message.LAT_LONG:
-                return ((MessageLatLong) message).getContent();
+            case Message.GEO:
+                return ((MessageGeo) message).getContent().getPointGeometry();
             case Message.IMAGE:
                 return ((MessageImage) message).getContent().getLocation();
             default:
-                throw new IllegalArgumentException("Message added to map should be supported type.");
```

Comment:
Do we really want to throw an exception on unsupported type?
I like to return null instead.